src/pybind/mgr/dashboard/controllers/home.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+import os
+import json
+
+import cherrypy
+from cherrypy.lib.static import serve_file
+
+from . import Controller, UiApiController, BaseController, Proxy, Endpoint
+from .. import mgr, logger
+
+
+LANGUAGES = {f for f in os.listdir(mgr.get_frontend_path()) \
+             if os.path.isdir("{}/{}".format(mgr.get_frontend_path(), f))}
+
+
+def _get_default_language():
+    with open("{}/../package.json".format(mgr.get_frontend_path()), "r") as f:
+        config = json.load(f)
+    return config['config']['locale']
+
+
+DEFAULT_LANGUAGE = _get_default_language()
+
+
+@Controller("/", secure=False)
+class HomeController(BaseController):
+
+    def _parse_accept_language(self):
+        if 'Accept-Language' in cherrypy.request.headers:
+            accept_lang_header = cherrypy.request.headers['Accept-Language']
+        else:
+            return [DEFAULT_LANGUAGE]
+
+        accept_langs = [l.strip() for l in accept_lang_header.split(",")]
+        result = []
+        for accept_lang in accept_langs:
+            if ';' in accept_lang:
+                lang = accept_lang.split(';')[0].strip()
```

Comment:
Let's ensure this is at least meeting RFC syntax (1*8ALPHA...). Otherwise, we are accepting here as a 'valid' language whatever an untrusted source is sending us.